tracing-attributes/src/expand.rs

```diff
@@ -14,6 +14,39 @@ use crate::{
     MaybeItemFn, MaybeItemFnRef,
 };
 
+#[derive(Debug, Default)]
+struct AsyncContext {
+    fake_return_edge: Option<proc_macro2::TokenStream>,
+}
```

Comment:
~~I'm not totally sure if I understand why `fake_return_edge` is an `Option` that's set to `Some` _after_ the struct is created using `Default` --- AFAICT, we always populate this field. It seems like the code could be a bit simpler if we changed this to just be~~
```suggestion
#[derive(Debug)]
struct AsyncContext {
    fake_return_edge: proc_macro2::TokenStream,
}
```

**EDIT**: wait, nevermind, I see where the `Option` is used --- we don't generate the fake return if the body is already an `async move` block rather than an `async fn`. It might be nice if there was a comment here explaining that, so it's clearer why this is optional?